spec/requests/api/v1/customers_spec.rb

```diff
@@ -105,6 +107,56 @@
       end
     end
 
+    describe "query parameters" do
+      describe "display_customer_balance" do
+        context "when true" do
+          let(:time) { "2023-01-01T00:00:00" }
+
+          it "adds balance at current time to each customer" do
+            allow(DateTime).to receive(:current).and_return(time)
+            get "/api/v1/customers", params: { display_customer_balance: "true" }
+            balances = json_response[:data].map{ |c| c[:attributes][:balance] }
+            all_times_current = balances.all?{ |b| b["time"].to_datetime == time.to_datetime }
+            all_values_numbers = balances.all?{ |b| b["value"] == 0 }
+            expect([all_times_current, all_values_numbers]).to eq([true, true])
+          end
+        end
+
+        context "when received non boolean" do
+          it "returns unprocessable entity" do
+            get "/api/v1/customers", params: { display_customer_balance: "unknown" }
+            expect(response.status).to eq 422
+          end
+        end
+
+        context "when not recevied" do
+          it "does not add balances" do
+            get "/api/v1/customers"
+            balances = json_response[:data].map{ |c| c[:attributes][:balance] }
+            expect(balances.compact).to be_empty
+          end
+        end
+      end
+
+      context "without display_customer_balance" do
+        let(:time) { "2023-01-01T00:00:00" }
+
+        it "adds balance at current time to each customer when true" do
+          allow(DateTime).to receive(:current).and_return(time)
+          get "/api/v1/customers", params: { display_customer_balance: "true" }
+          balances = json_response[:data].map{ |c| c[:attributes][:balance] }
+          all_times_current = balances.all?{ |b| b["time"].to_datetime == time.to_datetime }
+          all_values_numbers = balances.all?{ |b| b["value"] == 0 }
+          expect([all_times_current, all_values_numbers]).to eq([true, true])
+        end
+
+        it "returns unprocessable entity for non boolean values" do
+          get "/api/v1/customers", params: { display_customer_balance: "unknown" }
+          expect(response.status).to eq 422
```

Comment:
We may need to check the error message here because to make sure we are catching the right error.

Reply:
Fixed! Was surprised to see that method missing caused 422 instead of internal server error